packages/path_provider/path_provider_windows/lib/path_provider_windows.dart

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ffi';
+
+import 'package:ffi/ffi.dart';
+import 'package:win32/win32.dart';
+import 'package:path_provider_platform_interface/path_provider_platform_interface.dart';
+import 'package:path_provider_windows/folders.dart';
+
+/// The Windows implementation of [PathProviderPlatform]
+///
+/// This class implements the `package:path_provider` functionality for Windows
+class PathProviderWindows extends PathProviderPlatform {
+  /// Path to the temporary directory on the device that is not backed up and is
+  /// suitable for storing caches of downloaded files.
+  ///
+  /// On Windows, this the path specified by the TMP environment variable, or
+  /// the TEMP environment variable, or the USERPROFILE environment variable,
+  /// or the Windows directory, in order of preference. Windows does not
+  /// guarantee that the path exists or is writeable to.
```

Comment:
I think we should handle existence here; `path_provider` users should be shielded from the platform-specific details, and they are unlikey to come read these docs. Returning a temp path that doesn't exist seems like an unnecessary foot-gun for developers.